doc/manuals/contribution_guide.md

```diff
@@ -0,0 +1,674 @@
+  * **acceptance**: contains functional end-to-end test based on the Behave (NGSIv2) or Lettuce (NGSIv1) Python  
+    frameworks.
+* **scripts**: contains utility scripts (e.g. scripts included in the Orion RPM along with Orion binary itself,
+  scripts used by the test frameworks, etc.)
+* **doc**: contains documentation, with the following subdirectories:
+  * **apiary**: for apiary-based documentation
+  * **manuals**: for markdown based documentation
+* **rpm**: contains files for RPM building
+* **etc**: scripts that are installed under etc/ (typically, included in RPM package)
+* **docker**: contains the docker files
+* **archive**: contains older files that is no longer used but that we don’t want to remove yet.
+
+### File layout for source code files 
+
+Source code files are found under the `src/` directory.
+
+The suffix '.cpp' MUST be used for source files and '.h' MUST be used for header files.
+
+As a general rule, for C/C++ source code, every concept SHOULD have its own module. With a module is referred a 
+header file (`.h`) and its corresponding source file (`.cpp`). In some cases only a header file is needed (without any corresponding `.cpp` 
+source file): header files containing only constants, macros or inline functions.
+
+A class SHOULD reside in its own module, with the class definition in the header file
+and the implementation of the classes in the source file. The name for this module is the name of the class. E.g. 
+the class Subscription resides in the module { `Subscription.h` / `Subscription.cpp` }.
+
+If a class requires related classes (e.g. subclasses or embedded classes), they should reside in the same module, 
+but only if these classes are not used anywhere else. If used anywhere else, a new module MUST be created for the 
+affected classes.
+
+## Code style guidelines
+
+Note that currently not all Orion existing code base conforms to these rules. There are some parts of the code that were 
+written before the guidelines were established. However, all new code contributions MUST follow these rules and, eventually, old code will be modified to conform to the guidelines.
+
+### ‘MUST follow’ rules 
+
+#### M1 (Headers Files Inclusion):
+
+*Rule*: All header or source files MUST include all the header files it needs AND NO OTHER header files. They MUST
+NOT depend on inclusions of other header files. Also, all header and source files MUST NOT include any header files it 
+does not need itself.
+
+*Rationale*: each file should not depend on the inclusions other files have/doesn’t have. Also, if a header file
+includes more files than it needs, its ‘clients’ has no other choice than to include those ‘extra’ files as
+well. This sometimes leads to conflicts and must be avoided. In addition, it increases the compilation time.
+
+*How to check*": manually?, make lint?, other? TBD
+
+#### M2 (Copyright header)
+
+*Rule*: Every file, source code or not, MUST have a copyright header:
+
+For C++ files:
+
```

Comment:
perhaps we can have a template file and not display all this comment in this document